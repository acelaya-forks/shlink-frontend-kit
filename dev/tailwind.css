@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
