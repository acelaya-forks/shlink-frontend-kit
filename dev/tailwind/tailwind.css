@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
