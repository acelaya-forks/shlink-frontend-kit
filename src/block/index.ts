export * from './Message';
export * from './Result';
export * from './SimpleCard';
