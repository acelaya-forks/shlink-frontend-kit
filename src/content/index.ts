export * from './Details';
export * from './Listbox';
export * from './Table';
