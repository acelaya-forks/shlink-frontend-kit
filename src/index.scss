@import './utils/ResponsiveTable';
@import './theme/theme';
