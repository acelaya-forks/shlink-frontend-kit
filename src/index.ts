export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
