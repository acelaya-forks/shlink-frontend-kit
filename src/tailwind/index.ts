export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
