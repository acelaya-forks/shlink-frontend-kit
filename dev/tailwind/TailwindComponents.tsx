import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { DetailsPage } from './content/DetailsPage';
import { TablePage } from './content/TablePage';
import { MessagePage } from './feedback/MessagePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { TooltipPage } from './feedback/TooltipPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchComboboxPage } from './form/SearchComboboxPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';
import { UtilsPage } from './utils/UtilsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/search-combobox" element={<SearchComboboxPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/content/details" element={<DetailsPage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="/feedback/message" element={<MessagePage />} />
      <Route path="/feedback/tooltip" element={<TooltipPage />} />
      <Route path="/utils" element={<UtilsPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
