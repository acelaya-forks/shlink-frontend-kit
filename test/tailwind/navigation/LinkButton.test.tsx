import { render, screen } from '@testing-library/react';
import type { LinkButtonProps } from '../../../src/tailwind';
import { LinkButton } from '../../../src/tailwind';
import { checkAccessibility } from '../../__helpers__/accessibility';

describe('<LinkButton />', () => {
  const setUp = (props: LinkButtonProps = {}) => render(
    <div className="tw:bg-white">
      <LinkButton {...props} />
    </div>,
  );

  it('passes a11y checks', () => checkAccessibility(setUp({ children: 'Press me' })));

  it.each([
    {},
    { disabled: true },
    { size: 'sm' as const },
    { size: 'md' as const },
    { size: 'lg' as const },
  ])('renders as expected based on provided props', (props) => {
    setUp(props);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it.each([
    { type: undefined, expectedType: 'button' },
    { type: 'button' as const, expectedType: 'button' },
    { type: 'submit' as const, expectedType: 'submit' },
    { type: 'reset' as const, expectedType: 'reset' },
  ])('defaults type to `button`', ({ type, expectedType }) => {
    setUp({ type, children: 'The button' });
    expect(screen.getByRole('button', { name: 'The button' })).toHaveAttribute('type', expectedType);
  });
});
