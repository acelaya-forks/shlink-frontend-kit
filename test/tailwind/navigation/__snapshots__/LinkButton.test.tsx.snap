// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LinkButton /> > renders as expected based on provided props 1`] = `
<div>
  <button
    class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5"
  />
</div>
`;

exports[`<LinkButton /> > renders as expected based on provided props 2`] = `
<div>
  <button
    class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5 tw:pointer-events-none tw:opacity-65"
    disabled=""
  />
</div>
`;

exports[`<LinkButton /> > renders as expected based on provided props 3`] = `
<div>
  <button
    class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-1.5 tw:py-1 tw:text-sm"
  />
</div>
`;

exports[`<LinkButton /> > renders as expected based on provided props 4`] = `
<div>
  <button
    class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5"
  />
</div>
`;

exports[`<LinkButton /> > renders as expected based on provided props 5`] = `
<div>
  <button
    class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-4 tw:py-2 tw:text-lg"
  />
</div>
`;
